### PR TITLE
Fix phpstan complains about false errors

### PR DESCRIPTION
**Pull request by gmponos**

### CODE REVIEW

**Comment by gmponos** [comment]
For some weird reason PHPStan was complaining about these on master: https://github.com/guzzle/guzzle/runs/359402989

We had this PR merged before in order to fix them: https://github.com/guzzle/guzzle/pull/2471

also this error was removed: https://github.com/guzzle/guzzle/pull/2485/files#diff-cbcd91b5fa0a393ebd972f66cc47dc90L538

which initially it seems to have been cause again by an array_replace here: https://github.com/guzzle/guzzle/blob/master/src/Handler/StreamHandler.php#L298

The most interesting part is that I had this weird change in behavior locally too.. I am not sure if this is related in anyway with github actions or with PHPStan.. Also in GH actions I can not see the PHPStan version. So I am not sure what version it is used on each case...

File: src/Cookie/SetCookie.php [hunk 1]
```diff
@@ -69,12 +69,7 @@ public static function fromString(string $cookie): self
      */
     public function __construct(array $data = [])
     {
-        $replaced = \array_replace(self::$defaults, $data);
-        if ($replaced === null) {
```
Comment (sagikazarmark) [inline src/Cookie/SetCookie.php]: Why did you remove this check?

Comment (Nyholm) [reply src/Cookie/SetCookie.php]: If you revert this change, what will happen then?

Comment (gmponos) [reply src/Cookie/SetCookie.php]: Phpstan will complain :/

Comment (gmponos) [reply src/Cookie/SetCookie.php]: You can check master failing build.

Comment (sagikazarmark) [reply src/Cookie/SetCookie.php]: PHPStan probably things `array_replace` returns an array, not null. Removing the check is not a good idea IMHO, because the function *can* return null, in which case we should throw an exception.

Comment (gmponos) [reply src/Cookie/SetCookie.php]: OK.. I will leave the exception there.. My major problem here is that this error was reported two days ago.. and two days ago we merged this throw exception. What happened now with PHPStan and it false reports that array_replace returns only `array`?

Comment (gmponos) [reply src/Cookie/SetCookie.php]: > Removing the check is not a good idea IMHO

reverted..

**Comment by Nyholm** [comment]
CI is green, code is great.  =)
